### PR TITLE
DI-179 Log Invalid Postcode Messages To Splunk

### PR DESCRIPTION
**Pull request by rohith1122**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-179

## Description

Logging  Postcode validation failures to Splunk

### Noteworthy Changes

- Added  function to  log invalid postcode to Splunk
- Written unit test  for the above implementation 

## Type of change
- New feature (non-breaking change which adds functionality)

## Testing

Please tick the testing that has been completed

- [ ] Unit tests
- [ ] Component tests

## Developer Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the [code formatting checks](../README.md#code-quality)
- [x] I have run the [code quality checks](../README.md#code-quality)
- [x] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [x] Unit test code coverage is at or above 80%
- [x] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Component tests)
- [ ] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [x] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

File: application/event_processor/reporting.py
Comment (JackPlowman) [inline application/event_processor/reporting.py:66]: Missing second arg `dos_service`

File: application/event_processor/reporting.py
Comment (JackPlowman) [inline application/event_processor/reporting.py:61]: What your opinion to the suggestion below?

Can all functions in this file be renamed to a single standard and file name changed to `logging.py`.

**Review by JackPlowman — CHANGES_REQUESTED** [review]

File: application/event_processor/reporting.py
Comment (rohith1122) [reply application/event_processor/reporting.py:61]: Sounds good to me. I am happy to change it but there are a few places where we are using **log** and **report** for the same purpose in a few places.

File: application/event_processor/reporting.py
Comment (rohith1122) [reply application/event_processor/reporting.py:66]: Good spot. Thank you

File: application/event_processor/reporting.py
Comment (JackPlowman) [reply application/event_processor/reporting.py:61]: Yes, please standardising it now will be easier than we different people write new functions some with each standard

**Review by JackPlowman — COMMENTED** [review]
LGTM 👍 , Will approve once has been tested

**Review by JackPlowman — DISMISSED** [review]
LGTM 👍, Approving to remove the requested changes. Please don't merge until this ticket has been tested

**Comment by rohith1122** [comment]
> LGTM 👍, Approving to remove the requested changes. Please don't merge until this ticket has been tested

Sure

File: application/event_processor/dos.py
Comment (rohith1122) [reply application/event_processor/dos.py]: Sure.  Got it

**Review by mabe13 — APPROVED** [review]
LGTM 👍

**Review by SLClayton — APPROVED** [review]
thumbs up

**Pull request merged**
